cartographer/mapping/internal/3d/pose_graph_3d_test.cc

```diff
@@ -323,6 +323,59 @@ TEST_F(PoseGraph3DTest, EvenSubmapTrimmer) {
   }
 }
 
+TEST_F(PoseGraph3DTest, EvenSubmapTrimmerOnFrozenTrajectory) {
```

Comment:
Note: this test-case is based on the existing `EvenSubmapTrimmer` test case above, with the difference that no constraints are added and the graph is built like we do when loading frozen state.

There's no equivalent test case in `pose_graph_2d_test.cc` yet, so I did it only for 3D. If we want to extend the 2D test cases, this would be a larger change for separate PRs IMO.